sklearn/decomposition/_nmf.py

```diff
@@ -1163,6 +1163,13 @@ class NMF(TransformerMixin, BaseEstimator):
     max_iter : integer, default: 200
         Maximum number of iterations before timing out.
 
+    regularization : {'both', 'components', 'transformation', None}
+        Select whether the regularization affects the components (H), the
+        transformation (W), both or none of them.
+        Defaults to 'both'.
```

Comment:
The default needs to be `None` to be backward compatible, because the default of `regularization` in `non_negative_factorization` is `None`.

Reply:
Updated to `None`

Reply:
NMF used to call `non_negative_factorization` with `regularization='both'`, so the default should be 'both' for backward compat, right ?

Reply:
Ah yes, I was mistaken. Using `'both'` was correct.

Reply:
@Bharat123rox Sorry for the incorrect suggestion. The original implementation with `regularization='both'` as default was correct.